bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/helpers/TapoErrorHandler.java

```diff
@@ -90,27 +90,25 @@ public TapoErrorHandler(Exception ex, String infoMessage) {
     /**
      * GET ERROR-MESSAGE
      * 
-     * @param errCode error Number (or constant ERR_CODE )
-     * @return error-message if set constant ERR_CODE_MSG. if not name of ERR_CODE is returned
+     * @param errCode error Number (or constant ERR_API_CODE )
+     * @return error-message if set constant ERR_API_CODE_MSG. if not name of ERR_API_CODE is returned
      */
     private String getErrorMessage(Integer errCode) {
+        String i18nKey = "";
         Field[] fields = TapoErrorConstants.class.getDeclaredFields();
-        /* loop ErrorConstants and search for code in value */
+        /* loop ErrorConstants and search for integer-code in value */
         for (Field f : fields) {
             String constName = f.getName();
             try {
                 Integer val = (Integer) f.get(this);
                 if (val != null && val.equals(errCode)) {
-                    Field constantName = TapoErrorConstants.class.getDeclaredField(constName + "_MSG");
-                    String msg = getValueOrDefault(constantName.get(null), "").toString();
-                    if (msg.length() > 2) {
-                        return msg;
-                    } else {
-                        return infoMessage + " (" + constName + ")";
-                    }
+                    /* convert constatn name to i18n key */
+                    i18nKey = i18nKey.replaceFirst("ERR_", "error-");
+                    i18nKey = constName.replace("_", "-").toLowerCase();
```

Comment:
Looks like these two lines are in the wrong order.
With your current code, `i18nKey` is first a simple empty string so the first line will do nothing.

Reply:
Ok, I understood what you try to achieve.
I would then suggest only one line:
```
                     i18nKey = constName.replaceFirst("ERR_", "error-").replace("_", "-").toLowerCase();
```